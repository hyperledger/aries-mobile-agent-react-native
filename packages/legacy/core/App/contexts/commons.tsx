import { ConnectionRecord, CredentialExchangeRecord } from '@aries-framework/core'
import { useAgent } from '@aries-framework/react-hooks'
import React, { createContext, useContext } from 'react'

import { AppConsoleLogger, LogLevel } from '../services/logger'

export interface AppCommonsContext {
  log: (message: string, logLevel: Exclude<LogLevel, LogLevel.off>) => void
  getCredentialById(credentialRecordId: string): Promise<CredentialExchangeRecord>
  getConnectionById(connectionId: string): Promise<ConnectionRecord>
}

export const AppCommons = createContext<AppCommonsContext>(null as unknown as AppCommonsContext)

export const CommonUtilProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
  const logger = new AppConsoleLogger(LogLevel.test)
  const { agent } = useAgent()

  /** HOOKS */
  const log = async (messaage: string, logLevel: Exclude<LogLevel, LogLevel.off>) => {
    logger.log(logLevel, messaage)
    //TODO: do some more logic like collecting errors for later logging and investigation
  }

  /**
   * Retrieve a credential record by id
   *
   * @param credentialRecordId The credential record id
   * @return The credential record
   */
  async function getCredentialById(credentialRecordId: string): Promise<CredentialExchangeRecord> {
    try {
      if (!agent) {
        log(`[AppCommons]: Get credential by id: Agent not set`, LogLevel.error)
        throw new Error(`Agent not set`)
      }
      return await agent.credentials.getById(credentialRecordId)
    } catch (e: unknown) {
      log(`[AppCommons]: Get credential by id: ${e}`, LogLevel.error)
      throw new Error(`${e}`)
    }
  }

  /**
   * Retrieve a connection record by id
   *
   * @param connectionId The connection record id
   * @return The connection record
   */
  async function getConnectionById(connectionId: string): Promise<ConnectionRecord> {
    try {
      if (!agent) {
        log(`[AppCommons]: Get credential by id: Agent not set`, LogLevel.error)
        throw new Error(`Agent not set`)
      }
      return await agent.connections.getById(connectionId)
    } catch (e: unknown) {
      log(`[AppCommons]: Get getConnectionById by id: ${e}`, LogLevel.error)
      throw new Error(`${e}`)
    }
  }

  return (
    <AppCommons.Provider
      value={{
        log: log,
        getCredentialById: getCredentialById,
        getConnectionById: getConnectionById,
      }}
    >
      {children}
    </AppCommons.Provider>
  )
}

export const useCommons = () => {
  const commonsContext = useContext(AppCommons)
  if (!commonsContext) {
    throw new Error('commonsContext must be used within a CommonUtilProvider')
  }
  return commonsContext
}
