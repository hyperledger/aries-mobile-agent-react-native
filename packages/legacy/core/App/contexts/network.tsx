import { NetInfoStateType, useNetInfo } from '@react-native-community/netinfo'
import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import NetInfoModal from '../components/modals/NetInfoModal'
import { canConnectToLedgerNode } from '../utils/ledger'

export type LedgerNodeList = Array<{ host: string; port: number }>
export interface NetworkContext {
  silentAssertConnectedNetwork: () => boolean
  assertConnectedNetwork: () => boolean
  displayNetInfoModal: () => void
  hideNetInfoModal: () => void
  assertLedgerConnectivity: () => Promise<boolean | undefined>
  setLedgerNodes: (nodes: LedgerNodeList) => void
}

export const NetworkContext = createContext<NetworkContext>(null as unknown as NetworkContext)

export const NetworkProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
  const netInfo = useNetInfo()
  const [isNetInfoModalDisplayed, setIsNetInfoModalDisplayed] = useState<boolean>(false)
  let currentNodes: LedgerNodeList | undefined = undefined

  const setLedgerNodes = (nodes: LedgerNodeList) => {
    currentNodes = nodes
  }

  const displayNetInfoModal = () => {
    setIsNetInfoModalDisplayed(true)
  }

  const hideNetInfoModal = () => {
    setIsNetInfoModalDisplayed(false)
  }

  const silentAssertConnectedNetwork = () => {
    return netInfo.isConnected || netInfo.type !== NetInfoStateType.none
  }

  const assertConnectedNetwork = () => {
    const isConnected = silentAssertConnectedNetwork()
    if (!isConnected) {
      displayNetInfoModal()
    }

    return isConnected
  }

  const assertLedgerConnectivity = async (): Promise<boolean | undefined> => {
    if (typeof currentNodes === 'undefined') {
      return undefined
    }

    if (currentNodes.length === 0) {
      return false
    }

    const connections = await Promise.all(
      currentNodes.map((n: { host: string; port: number }) => canConnectToLedgerNode(n))
    )

    return connections.includes(true)
  }

  return (
    <NetworkContext.Provider
      value={{
        silentAssertConnectedNetwork,
        assertConnectedNetwork,
        displayNetInfoModal,
        hideNetInfoModal,
        assertLedgerConnectivity,
        setLedgerNodes,
      }}
    >
      {children}
      <NetInfoModal visible={isNetInfoModalDisplayed} onSubmit={() => hideNetInfoModal()} />
    </NetworkContext.Provider>
  )
}

export const useNetwork = () => useContext(NetworkContext)
