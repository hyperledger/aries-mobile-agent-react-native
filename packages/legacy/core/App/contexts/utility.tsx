import React, { createContext, useContext } from 'react'
import { BaseLogger } from '@credo-ts/core'

import { ConsoleLogger } from '../services/logger'

export interface UtilityContext {
  log: BaseLogger
}

export const Utility = createContext<UtilityContext>(null as unknown as UtilityContext)

export const UtilityProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
  const logger = new ConsoleLogger()

  return (
    <Utility.Provider
      value={{
        log: logger,
      }}
    >
      {children}
    </Utility.Provider>
  )
}

export const useUtility = () => {
  const utilityContext = useContext(Utility)
  if (!utilityContext) {
    throw new Error('utilityContext must be used within a CommonUtilProvider')
  }

  return utilityContext
}
