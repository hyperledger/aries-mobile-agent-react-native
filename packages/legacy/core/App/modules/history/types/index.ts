export enum HistoryCardType {
  CardAccepted = 'CardAccepted',
  CardDeclined = 'CardDeclined',
  CardExpired = 'CardExpired',
  CardRevoked = 'CardRevoked',
  InformationSent = 'InformationSent',
  PinChanged = 'PinChanged',
  CardUpdates = 'CardUpdates',
}
export interface HistoryRecord {
  /** History Record ID */
  id?: string
  /** History Record Type */
  type?: HistoryCardType
  /** History Record Message */
  message: string
  /** History Record Date */
  createdAt?: Date
  /** History Record Item ID */
  correspondenceId?: string
  /** History Record Item Name */
  correspondenceName?: string
}
export enum RecordType {
  AppNotificationRecord = 'AppNotificationRecord',
  NotificationRecord = 'NotificationRecord',
  HistoryRecord = 'HistoryRecord',
}
export type TagObject = {
  [key: string]: string | boolean | undefined | Array<string> | null
  [key: number]: never
}

export interface CustomRecord {
  content: RecordTypes
  tags?: TagObject
}

export interface AppNotificationRecord {
  id?: string
  createdAt?: Date
  type?: AppNotificationCardType
}

export interface NotificationRecord {
  message: string
  credentialId: string
  type?: NotificationCardType
  id?: string
  createdAt?: Date
  isNotificationRemoved: boolean
}

export enum AppNotificationCardType {
  AppPinReminder = 'AppPinReminder',
}

export enum NotificationCardType {
  CardExpired = 'CardExpired',
  CardRevoked = 'CardRevoked',
}

type RecordTypes = NotificationRecord | HistoryRecord | AppNotificationRecord
export type HistoryQuery = {
  type: RecordType
}
