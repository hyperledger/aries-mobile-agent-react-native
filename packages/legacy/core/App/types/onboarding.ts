export interface OnboardingStyleSheet {
  container: Record<string, any>
  carouselContainer: Record<string, any>
  pagerContainer: Record<string, any>
  pagerDot: Record<string, any>
  pagerDotActive: Record<string, any>
  pagerDotInactive: Record<string, any>
  pagerPosition: Record<string, any>
  pagerNavigationButton: Record<string, any>
}
