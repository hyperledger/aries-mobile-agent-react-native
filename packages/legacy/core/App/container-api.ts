import { StackNavigationProp } from '@react-navigation/stack'
import { createContext, useContext } from 'react'

import { Button } from './components/buttons/Button-api'
import { ReducerAction } from './contexts/reducers/store'
import Onboarding from './screens/Onboarding'
import { GenericFn } from './types/fn'
import { AuthenticateStackParams } from './types/navigators'

export enum SCREEN_TOKENS {
  SCREEN_TERMS = 'screen.terms',
  SCREEN_ONBOARDING = 'screen.onboarding',
  SCREEN_ONBOARDING_ITEM = 'screen.onboarding.item',
}
export enum STACK_TOKENS {
  STACK_ONBOARDING = 'stack.onboarding',
}
export enum FN_TOKENS {
  FN_ONBOARDING_DONE = 'fn.onboardingDone',
}

export enum COMP_TOKENS {
  COMP_BUTTON = 'comp.button',
}

export enum SERVICE_TOKENS {
  SERVICE_TERMS = 'screen.terms',
}

export const TOKENS = { ...SCREEN_TOKENS, ...SERVICE_TOKENS, ...STACK_TOKENS, ...FN_TOKENS, ...COMP_TOKENS }

export type FN_ONBOARDING_DONE = (
  dispatch: React.Dispatch<ReducerAction<unknown>>,
  navigation: StackNavigationProp<AuthenticateStackParams>
) => GenericFn

export interface TokenMapping {
  [TOKENS.STACK_ONBOARDING]: React.FC
  [TOKENS.SCREEN_TERMS]: React.FC
  [TOKENS.SCREEN_ONBOARDING]: typeof Onboarding
  [TOKENS.FN_ONBOARDING_DONE]: FN_ONBOARDING_DONE
  [TOKENS.COMP_BUTTON]: Button
}

export interface System {
  init(): System
  resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K]
}

export const SystemContext = createContext<System | undefined>(undefined)

export const SystemProvider = SystemContext.Provider

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
export const useContainer = () => useContext(SystemContext)!
