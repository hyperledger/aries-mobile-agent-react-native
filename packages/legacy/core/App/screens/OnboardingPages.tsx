import React from 'react'
import { useTranslation } from 'react-i18next'
import { ScrollView, StyleSheet, Text, View } from 'react-native'

import CredentialList from '../assets/img/credential-list.svg'
import ScanShare from '../assets/img/scan-share.svg'
import SecureImage from '../assets/img/secure-image.svg'
import Button, { ButtonType } from '../components/buttons/Button'
import { GenericFn } from '../types/fn'
import { OnboardingStyleSheet } from '../types/onboarding'
import { testIdWithKey } from '../utils/testable'

export const createCarouselStyle = (OnboardingTheme: any): OnboardingStyleSheet => {
  return StyleSheet.create({
    container: {
      ...OnboardingTheme.container,
      flex: 1,
      alignItems: 'center',
    },
    carouselContainer: {
      ...OnboardingTheme.carouselContainer,
      flexDirection: 'column',
    },
    pagerContainer: {
      flexShrink: 1,
      flexDirection: 'row',
      alignItems: 'center',
      marginBottom: 30,
    },
    pagerDot: {
      ...OnboardingTheme.pagerDot,
      borderWidth: 1,
      borderStyle: 'solid',
    },
    pagerDotActive: {
      ...OnboardingTheme.pagerDotActive,
    },
    pagerDotInactive: {
      ...OnboardingTheme.pagerDotInactive,
    },
    pagerPosition: {
      position: 'relative',
      top: 0,
    },
    pagerNavigationButton: {
      ...OnboardingTheme.pagerNavigationButton,
      fontSize: 18,
      fontWeight: 'bold',
    },
  })
}

export const createStyles = (OnboardingTheme: any) => {
  return StyleSheet.create({
    headerText: {
      ...OnboardingTheme.headerText,
    },
    bodyText: {
      ...OnboardingTheme.bodyText,
      flexShrink: 1,
    },
    point: {
      flexDirection: 'row',
      alignItems: 'center',
      marginLeft: 20,
      marginTop: 10,
      marginRight: 20,
      marginBottom: 10,
    },
    icon: {
      marginRight: 10,
    },
  })
}

const createImageDisplayOptions = (OnboardingTheme: any) => {
  return {
    ...OnboardingTheme.imageDisplayOptions,
    height: 180,
    width: 180,
  }
}

const firstPage = (OnboardingTheme: any) => {
  const { t } = useTranslation()
  const styles = createStyles(OnboardingTheme)
  const imageDisplayOptions = createImageDisplayOptions(OnboardingTheme)
  return (
    <ScrollView style={{ padding: 20 }}>
      <View style={{ alignItems: 'center' }}>
        <CredentialList {...imageDisplayOptions} />
      </View>
      <View style={{ marginBottom: 20 }}>
        <Text style={[styles.headerText, { fontSize: 18 }]} testID={testIdWithKey('HeaderText')}>
          {t('Onboarding.Page1Title')}
        </Text>
        <Text style={[styles.bodyText, { marginTop: 25 }]} testID={testIdWithKey('BodyText')}>
          {t('Onboarding.Page1Body')}
        </Text>
      </View>
    </ScrollView>
  )
}

const secondPage = (OnboardingTheme: any) => {
  const { t } = useTranslation()
  const styles = createStyles(OnboardingTheme)
  const imageDisplayOptions = createImageDisplayOptions(OnboardingTheme)
  return (
    <ScrollView style={{ padding: 20 }}>
      <View style={{ alignItems: 'center' }}>
        <ScanShare {...imageDisplayOptions} />
      </View>
      <View style={{ marginBottom: 20 }}>
        <Text style={[styles.headerText, { fontSize: 18 }]} testID={testIdWithKey('HeaderText')}>
          {t('Onboarding.Page2Title')}
        </Text>
        <Text style={[styles.bodyText, { marginTop: 25 }]} testID={testIdWithKey('BodyText')}>
          {t('Onboarding.Page2Body')}
        </Text>
      </View>
    </ScrollView>
  )
}

const thirdPage = (onTutorialCompleted: GenericFn, OnboardingTheme: any) => {
  const { t } = useTranslation()
  const styles = createStyles(OnboardingTheme)
  const imageDisplayOptions = createImageDisplayOptions(OnboardingTheme)
  return (
    <>
      <ScrollView style={{ padding: 20 }}>
        <View style={{ alignItems: 'center' }}>
          <SecureImage {...imageDisplayOptions} />
        </View>
        <View style={{ marginBottom: 20 }}>
          <Text style={[styles.headerText, { fontSize: 18 }]} testID={testIdWithKey('HeaderText')}>
            {t('Onboarding.Page3Title')}
          </Text>
          <Text style={[styles.bodyText, { marginTop: 25 }]} testID={testIdWithKey('BodyText')}>
            {t('Onboarding.Page3Body')}
          </Text>
        </View>
      </ScrollView>
      <View style={{ marginTop: 'auto', margin: 20 }}>
        <Button
          title={t('Global.GetStarted')}
          accessibilityLabel={t('Global.GetStarted')}
          testID={testIdWithKey('GetStarted')}
          onPress={onTutorialCompleted}
          buttonType={ButtonType.Primary}
        />
      </View>
    </>
  )
}

const OnboardingPages = (onTutorialCompleted: GenericFn, OnboardingTheme: any): Array<Element> => {
  return [firstPage(OnboardingTheme), secondPage(OnboardingTheme), thirdPage(onTutorialCompleted, OnboardingTheme)]
}

export default OnboardingPages
