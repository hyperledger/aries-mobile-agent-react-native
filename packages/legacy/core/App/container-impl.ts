import { StackNavigationProp } from '@react-navigation/stack'
import { createContext, useContext } from 'react'
import { DependencyContainer } from 'tsyringe'

import Button from './components/buttons/Button'
import { TOKENS, System, TokenMapping } from './container-api'
import { DispatchAction, ReducerAction } from './contexts/reducers/store'
import OnboardingStack from './navigators/OnboardingStack'
import Onboarding from './screens/Onboarding'
import ScreenTerms from './screens/Terms'
import { AuthenticateStackParams, Screens } from './types/navigators'

export class MainSystem implements System {
  public static readonly TOKENS = TOKENS
  private container: DependencyContainer
  public constructor(container: DependencyContainer) {
    this.container = container
  }
  public init(): System {
    // eslint-disable-next-line no-console
    console.log(`initializing System`)
    this.container.registerInstance(TOKENS.SCREEN_TERMS, ScreenTerms)
    this.container.registerInstance(TOKENS.SCREEN_ONBOARDING, Onboarding)
    this.container.registerInstance(TOKENS.STACK_ONBOARDING, OnboardingStack)
    this.container.registerInstance(TOKENS.COMP_BUTTON, Button)

    this.container.registerInstance(
      TOKENS.FN_ONBOARDING_DONE,
      (dispatch: React.Dispatch<ReducerAction<unknown>>, navigation: StackNavigationProp<AuthenticateStackParams>) => {
        return () => {
          dispatch({
            type: DispatchAction.DID_COMPLETE_TUTORIAL,
          })
          navigation.navigate(Screens.Terms)
        }
      }
    )
    return this
  }
  public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
    // eslint-disable-next-line no-console
    console.log(`resolving ${token}`)
    return this.container.resolve(token) as TokenMapping[K]
  }
}

export const SystemContext = createContext<System | undefined>(undefined)

export const SystemProvider = SystemContext.Provider

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
export const useSystem = () => useContext(SystemContext)!
