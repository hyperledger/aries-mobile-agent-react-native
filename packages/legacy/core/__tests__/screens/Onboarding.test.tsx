import { render } from '@testing-library/react-native'
import React from 'react'

import * as themeContext from '../../App/contexts/theme' // note we're importing with a * to import all the exports
import Onboarding from '../../App/screens/Onboarding'
import { createCarouselStyle } from '../../App/screens/OnboardingPages'
import { OnboardingTheme, theme } from '../../App/theme'
import { OnboardingStyleSheet } from '../../App/types/onboarding'

export const carousel: OnboardingStyleSheet = createCarouselStyle(OnboardingTheme)

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

describe('Onboarding', () => {
  test('Renders correctly', () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const tree = render(
      <Onboarding onTutorialCompleted={() => null} nextButtonText="Next" previousButtonText="Back" style={carousel} />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Pages exist', async () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const { findAllByTestId } = render(
      <Onboarding onTutorialCompleted={() => null} nextButtonText="Next" previousButtonText="Back" style={carousel} />
    )
    const foundPages = await findAllByTestId('BodyText', { exact: false })

    expect(foundPages.length).toBe(3)
  })
})
