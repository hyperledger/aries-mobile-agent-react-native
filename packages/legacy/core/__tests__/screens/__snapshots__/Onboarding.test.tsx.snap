// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        <ScrollView
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <
              fill="#FFFFFF"
              height={180}
              width={180}
            />
          </View>
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 32,
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 18,
                  },
                ]
              }
              testID="com.ariesbifold:id/HeaderText"
            >
              Onboarding.Page1Title
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginTop": 25,
                  },
                ]
              }
              testID="com.ariesbifold:id/BodyText"
            >
              Onboarding.Page1Body
            </Text>
          </View>
        </ScrollView>,
        <ScrollView
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <
              fill="#FFFFFF"
              height={180}
              width={180}
            />
          </View>
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 32,
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 18,
                  },
                ]
              }
              testID="com.ariesbifold:id/HeaderText"
            >
              Onboarding.Page2Title
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginTop": 25,
                  },
                ]
              }
              testID="com.ariesbifold:id/BodyText"
            >
              Onboarding.Page2Body
            </Text>
          </View>
        </ScrollView>,
        <React.Fragment>
          <ScrollView
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <
                fill="#FFFFFF"
                height={180}
                width={180}
              />
            </View>
            <View
              style={
                Object {
                  "marginBottom": 20,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 32,
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontSize": 18,
                    },
                  ]
                }
                testID="com.ariesbifold:id/HeaderText"
              >
                Onboarding.Page3Title
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "flexShrink": 1,
                      "fontSize": 18,
                      "fontWeight": "normal",
                    },
                    Object {
                      "marginTop": 25,
                    },
                  ]
                }
                testID="com.ariesbifold:id/BodyText"
              >
                Onboarding.Page3Body
              </Text>
            </View>
          </ScrollView>
          <View
            style={
              Object {
                "margin": 20,
                "marginTop": "auto",
              }
            }
          >
            <ForwardRef
              accessibilityLabel="Global.GetStarted"
              buttonType={1}
              onPress={[Function]}
              testID="com.ariesbifold:id/GetStarted"
              title="Global.GetStarted"
            />
          </View>
        </React.Fragment>,
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "width": 750,
        },
      ]
    }
    viewabilityConfig={
      Object {
        "viewAreaCoveragePercentThreshold": 60,
      }
    }
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": Object {
            "viewAreaCoveragePercentThreshold": 60,
          },
        },
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 750,
              },
              Object {
                "backgroundColor": "#000000",
                "flexDirection": "column",
              },
            ]
          }
        >
          <RCTScrollView
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <
                  fill="#FFFFFF"
                  height={180}
                  width={180}
                />
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 32,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontSize": 18,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/HeaderText"
                >
                  Onboarding.Page1Title
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "flexShrink": 1,
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {
                        "marginTop": 25,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  Onboarding.Page1Body
                </Text>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 750,
              },
              Object {
                "backgroundColor": "#000000",
                "flexDirection": "column",
              },
            ]
          }
        >
          <RCTScrollView
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <
                  fill="#FFFFFF"
                  height={180}
                  width={180}
                />
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 32,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontSize": 18,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/HeaderText"
                >
                  Onboarding.Page2Title
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "flexShrink": 1,
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {
                        "marginTop": 25,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  Onboarding.Page2Body
                </Text>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 750,
              },
              Object {
                "backgroundColor": "#000000",
                "flexDirection": "column",
              },
            ]
          }
        >
          <RCTScrollView
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <
                  fill="#FFFFFF"
                  height={180}
                  width={180}
                />
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 32,
                        "fontWeight": "bold",
                      },
                      Object {
                        "fontSize": 18,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/HeaderText"
                >
                  Onboarding.Page3Title
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "flexShrink": 1,
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {
                        "marginTop": 25,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  Onboarding.Page3Body
                </Text>
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              Object {
                "margin": 20,
                "marginTop": "auto",
              }
            }
          >
            <View
              accessibilityLabel="Global.GetStarted"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#42803E",
                  "borderRadius": 4,
                  "opacity": 1,
                  "padding": 16,
                }
              }
              testID="com.ariesbifold:id/GetStarted"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                      false,
                    ]
                  }
                >
                  Global.GetStarted
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "marginBottom": 30,
      }
    }
  >
    <View
      accessibilityLabel="Global.Back"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Back"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#42803E",
              "fontSize": 18,
              "fontWeight": "bold",
            },
            Object {
              "color": "transparent",
              "paddingRight": 20,
            },
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 20,
            "flexDirection": "row",
            "position": "absolute",
          },
          Object {
            "position": "relative",
            "top": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "rgba(66, 128, 62, 1)",
            "borderColor": "#42803E",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 10,
            "marginHorizontal": 5,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderColor": "#42803E",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 10,
            "marginHorizontal": 5,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderColor": "#42803E",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 10,
            "marginHorizontal": 5,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="Global.Next"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Next"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#42803E",
              "fontSize": 18,
              "fontWeight": "bold",
            },
            Object {
              "color": "#42803E",
              "paddingLeft": 20,
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
